### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by erikagayan**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,2 +1,71 @@
...
+    @property
+    def current_max_size(self) -> float:
+        return RESIZE_THRESHOLD * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
```
Comment (Oleg-s-s) [inline app/main.py:30]: No need to write `is not None` here
Fix everywhere

File: app/main.py [hunk 2]
```diff
@@ -1,2 +1,71 @@
...
+
+        return index
+
+    def _cyclic_increment(self, index: int) -> int:
+        return (index + 1) % self.capacity
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
```
Comment (Oleg-s-s) [inline app/main.py:41]: No need to write `is None` 
Fix everywhere

**Review by Oleg-s-s — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,2 +1,71 @@
...
+        if not self.hash_table[index]:
+            raise KeyError(f"Cannot find value for key: {key}")
+
+        return self.hash_table[index].value
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._calculate_index(key)
+
+        if not self.hash_table[index]:  # add new element
+            if self.size + 1 >= self.current_max_size:
```
Comment (LiudmylaKulinchenko) [inline app/main.py:50]: You can remove comments

File: app/main.py [hunk 4]
```diff
@@ -1,2 +1,71 @@
...
+
+INITIAL_CAPACITY = 8
+RESIZE_THRESHOLD = 2 / 3
+CAPACITY_MULTIPLIER = 2
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
```
Comment (LiudmylaKulinchenko) [inline app/main.py:12]: You should also store hash from a key value in Node

**Review by Oleg-s-s — APPROVED** [review]
:+1:

**Pull request closed without merging**